### PR TITLE
8282915: [vectorapi] Add new vector operation for reverse bits operation

### PR DESCRIPTION
**Pull request by smita-kamath**
Hi,

I've added support for new vector operations for reversing bits of integral vector types(Byte/Short/Integer/Long).
Currently, this PR addresses only Java changes for reverse operation. I've also updated the test framework.
Do review and share feedback.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282915](https://bugs.openjdk.java.net/browse/JDK-8282915): [vectorapi] Add new vector operation for reverse bits operation

### Reviewers
 * [Paul Sandoz](https://openjdk.java.net/census#psandoz) (@PaulSandoz - Committer) ⚠️ Review applies to 61d6af27acc5c319347d2fb33893ce8904a09407

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-vector pull/178/head:pull/178` \
`$ git checkout pull/178`

Update a local copy of the PR: \
`$ git checkout pull/178` \
`$ git pull https://git.openjdk.java.net/panama-vector pull/178/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 178`

View PR using the GUI difftool: \
`$ git pr show -t 178`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-vector/pull/178.diff">https://git.openjdk.java.net/panama-vector/pull/178.diff</a>

</details>

### CODE REVIEW

File: test/jdk/jdk/incubator/vector/templates/Unit-header.template [hunk 1]
```diff
@@ -1481,6 +1481,26 @@ public class $vectorteststype$ extends AbstractVectorTest {
...
+#else[short]
+        $type$ b = ($type$) ROL_scalar(a, ($type$) 4);
+        b = (byte)(((b & 0x55) << 1) | ((b & 0xAA) >>> 1));
+        b = (byte)(((b & 0x33) << 2) | ((b & 0xCC) >>> 2));
+        return b;
+#end[short]
+#end[intOrLong]
+    }
+
+
```
Comment (XiaohongGong) [inline test/jdk/jdk/incubator/vector/templates/Unit-header.template:1503]: Remove one extra line?

**Comment by jatin-bhateja** [comment]
Can we introduce special handling for 0 and -1 value ? in this case Java side fallback implementation can simply return the original argument.
Apart from above and spacing comments patch looks good to me.

File: test/jdk/jdk/incubator/vector/templates/Unit-header.template (hunk 1 above)
Comment (smita-kamath) [reply test/jdk/jdk/incubator/vector/templates/Unit-header.template:1503]: Will fix it. Thank you.

**Review by PaulSandoz — APPROVED** [review]

**Comment by merykitty** [comment]
I suggest being more explicit about the operation name because there is `reverseBytes` as well.

**Comment by smita-kamath** [comment]
@merykitty we have reverseBytes operation in development. I was following the naming convention similar to whats in Integer class (which has Integer.reverse operation).

**Comment by XiaohongGong** [comment]
Hi @smita-kamath , do you plan to add the hotspot compiler support for the new added API? Thanks so much!

**Pull request closed without merging**